### PR TITLE
feat(platform/dashboard): Enable editing for agent submissions

### PR DESCRIPTION
**Pull request by Abhi1992002**
- resolves - https://github.com/Significant-Gravitas/AutoGPT/issues/10511

In this PR, I’ve added backend endpoints and a frontend UI for edit functionality on the Agent Dashboard. Now, users can update their store submission, if status is `PENDING` or `APPROVED`, but not for `REJECTED` and `DRAFT`. When users make changes to a pending status submission, the changes are made to the same version. However, when users make changes to an approved status submission, a new store listing version is created.

Backend works something like this: 

Screenshot 2025-08-15 at 9 39 02 AM

### Changes
- I’ve updated the `StoreSubmission` view to include `video_url` and `categories`.
- I’ve added a new frontend UI for editing submissions.
- I’ve created an endpoint for editing submissions.
- I’ve added more end-to-end tests to ensure the edit submission functionality works as expected.

#### For code changes:
- [x] I have clearly listed my changes in the PR description
- [x] I have made a test plan
- [x] I have tested my changes according to the test plan:
  - [x] I have checked manually, everything is working perfectly.
  - [x] All e2e tests are also passing.

### CODE REVIEW

**Comment by AutoGPT-Agent** [comment]
Thank you for your work on enabling editing for agent submissions. This feature will be a valuable addition, but there are a few things that need to be addressed before this PR can be considered for merging:

1. The PR description is missing important details about what changes you've made and why they're needed. Please fill in the "Changes" section with a clear explanation of what you've implemented.

2. None of the checklist items have been completed. Please check off the items that apply to your changes and include a test plan that describes how you've tested this new functionality.

3. The PR title should follow our conventional commit format. Consider changing it from "WIP: Enable Editing for Agent Submissions" to something like "feat(platform/marketplace): Enable editing for agent submissions".

4. Remove the "WIP" (Work in Progress) prefix from the title when you're ready for review.

From a code perspective, the implementation looks good - you've added proper user_id checks in your new functions and handled the different submission states appropriately. Once you've addressed the PR formatting issues, this should be ready for a more thorough review.

**Comment by AutoGPT-Agent** [comment]
Thanks for your PR to enable editing for agent submissions! This looks like a valuable feature that will improve the marketplace experience. However, there are a few issues that need to be addressed before this can be merged:

## Required Changes

1. **PR Title**: Please update the title to follow our conventional commit format. For example: `feat(platform/marketplace): Enable Editing for Agent Submissions`

2. **PR Description**: Please complete the PR description to explain:
   - The purpose of these changes
   - A concise summary of what was changed (backend routes, frontend components, etc.)

3. **Checklist**: The checklist must be filled out completely. Please:
   - Check off each item that applies
   - Add a test plan outlining how you've tested this new functionality

## Code Implementation

The implementation looks solid! I particularly like how you:
- Added proper user_id checks for security
- Created a clean separation between editing PENDING vs. APPROVED submissions
- Reused components where possible
- Added appropriate error handling

Once you've addressed the PR format issues above, this should be ready for another review. Let me know if you have any questions!

**Comment by AutoGPT-Agent** [comment]
Thank you for your PR to enable editing for agent submissions. This looks like a valuable feature that will improve user experience. However, I cannot approve this PR in its current state for the following reasons:

1. Your PR is marked as "WIP" (Work in Progress), suggesting it's not ready for review yet.

2. The PR description is completely empty - you need to clearly explain the purpose and changes made in this PR.

3. None of the checklist items have been completed. For code changes, you need to:
   - List your changes in the PR description
   - Create and execute a test plan
   - Fill out the appropriate checklist items

The code itself looks well-structured and implements a complete edit functionality for marketplace submissions, including:
- Backend functionality to retrieve and update submissions
- New exception types for handling errors
- Frontend components for the edit UI
- API route updates

Please update your PR with:
1. A proper title (without the "WIP:" prefix)
2. A clear description of what this feature does and why it's needed
3. A completed checklist including your test plan

Once these updates are made, we can proceed with a proper review of your code changes.

**Comment by AutoGPT-Agent** [comment]
# PR Review - WIP: Enable Editing for Agent Submissions

Thank you for your work on adding functionality to edit agent submissions. However, this PR needs significant updates before it can be considered for merging.

## Major issues:

1. **Incomplete PR Description**
   - Your PR is missing a description explaining what changes you've made and why
   - None of the checklist items are checked off
   - No test plan is provided

2. **PR Title Format**
   - The title starts with "WIP" indicating it's not ready for review
   - The title doesn't follow conventional commit format (should be something like `feat(platform/frontend): Enable editing for agent submissions`)

3. **Implementation Concerns**
   - The backend changes include a new `edit_store_submission` function that accepts a user_id, but needs proper validation to ensure the user only edits their own submissions
   - The frontend code would benefit from more robust error handling and user feedback

## What needs to be done:

1. **Complete the PR description**:
   - Explain the purpose and scope of these changes
   - Fill out the checklist completely
   - Add a detailed test plan explaining how to test the editing functionality

2. **Update the PR title**:
   - Remove the "WIP" prefix when ready for review
   - Follow the conventional commit format with appropriate type and scope

3. **Code improvements**:
   - Add proper comments explaining the user_id validation in the backend code
   - Ensure consistent error handling across all new functionality
   - Consider adding tests for the new functionality

Please address these issues and update the PR when it's ready for a full review. I'm happy to provide more detailed feedback on the implementation once the PR description and structure are updated.

**Comment by AutoGPT-Agent** [comment]
Thank you for your contribution to enable agent submission editing. This is an important feature, but your PR needs some updates before it can be merged:

### PR Metadata Issues

1. **Title Format**: Please update your PR title to follow our conventional commit format. Based on your changes, something like `feat(platform/marketplace): enable editing for agent submissions` would be appropriate.

2. **Description**: Please add a detailed description of what this PR does and why it's needed. Currently, the description section is empty.

3. **Checklist**: None of the checklist items are completed. Please fill out the checklist appropriately, including your test plan.

### Code Review

The code implementation looks good overall. You've added:

- A new backend function `edit_store_submission` with proper user_id checking
- Frontend components for editing agent submissions
- Database migration for updated store submission view
- New model types to support the edit functionality

Some specific observations:

- The backend code correctly handles different submission statuses (PENDING vs APPROVED)
- User permissions are properly checked in the backend functions
- The frontend modal and form components look well structured

Once you've updated the PR title, added a description, and completed the checklist (especially the test plan), this PR should be ready for another review.

As this is marked as WIP, I assume you're still making changes, but these updates will be necessary before the PR can be merged.

**Comment by AutoGPT-Agent** [comment]
Thanks for your work on enabling editing for agent submissions. This looks like a useful feature, but your PR needs some updates before it can be considered for merging:

### Required Changes

1. **Remove the WIP prefix**: Once your PR is ready for review, update the title to remove "WIP:" and follow the conventional commit format like `feat(platform/frontend): Enable Editing for Agent Submissions`

2. **Complete the PR description**: Please explain:
   - Why this feature is needed
   - What specific changes you've made
   - Any architectural decisions worth noting

3. **Fill out the checklist**: Check off all applicable items in the PR checklist, especially:
   - Changes description
   - Test plan (how have you verified this works?)
   - Any configuration changes

### Technical Feedback

The implementation looks solid overall. You've:
- Added proper backend functionality with `edit_store_submission`
- Created appropriate error handling
- Added UI components for editing submissions
- Updated the types and API models

Once you've completed the PR description and checklist, and updated the title to follow our conventional commit format, this should be ready for a proper review.

Please let me know if you have any questions!

**Comment by AutoGPT-Agent** [comment]
# PR Review Feedback

Thank you for working on the agent submission editing functionality. This is a valuable feature that will improve the marketplace experience. However, there are several issues that need to be addressed before this PR can be merged:

## Required Changes

1. **PR Description and Checklist**
   - Please provide a complete description of the changes in this PR, explaining what problem you're solving and how your implementation works
   - Fill out the checklist in the PR description, including your test plan

2. **PR Title Format**
   - The title should follow conventional commit format. Instead of "WIP: Enable Editing for Agent Submissions", use something like:
   - "feat(platform/marketplace): Enable editing for agent submissions"

3. **Test Coverage**
   - Include a test plan detailing how you've verified the functionality works as expected
   - Consider edge cases like what happens when attempting to edit:
     - A rejected submission
     - An approved submission (creates a new version)
     - A pending submission (updates existing version)

## Technical Implementation

The implementation looks generally good, with proper user ID checks in backend functions and appropriate error handling. I particularly like:

- The conditional logic that handles different submission states (PENDING vs APPROVED) differently
- The appropriate user authorization checks
- The frontend implementation that reuses much of the existing submission components

Once you've addressed the PR description and checklist requirements, this looks ready for a technical review.

**Comment by AutoGPT-Agent** [comment]
Thanks for the PR @Abhi1992002. This looks like a valuable addition to enable users to edit their agent submissions. However, there are several items that need to be addressed before we can merge this:

### Required Changes

1. **PR Title**: Please update the title to follow our conventional commit format. Since this is a feature for the marketplace platform, something like: `feat(platform/marketplace): Enable Editing for Agent Submissions` would be appropriate.

2. **PR Description**: Please complete the PR description by:
   - Explaining the need for these changes
   - Describing the specific changes you've made
   - Checking off the items in the checklist that you've completed

3. **Test Plan**: Please include a test plan that outlines how you've tested these changes.

### Technical Review Notes

- The backend implementation looks solid with proper user authorization checks in the `edit_store_submission` function
- The frontend components are well-structured to support the editing workflow
- The database migration for adding video_url and categories to the store submission view is correctly implemented

Before this can be merged, please:
1. Complete the PR description and checklist
2. Update the title to follow our conventional format
3. Remove the WIP status when you feel the PR is ready for final review

Let me know if you need any help with these changes.

**Comment by AutoGPT-Agent** [comment]
Thanks for your PR implementing agent submission editing functionality! The code changes look thorough and well-implemented, but there are some issues that need to be addressed before this can be merged:

### PR Description Issues

- Please complete the PR description to explain the purpose of these changes and why they're needed
- Fill out the checklist completely, particularly:
  - List your specific changes
  - Provide a test plan detailing how you've tested this functionality

### Technical Review

The code implementation looks good! I particularly like:

- Thorough implementation of the backend edit_store_submission function with proper user ownership validation
- Good error handling with specific exception types
- Comprehensive frontend implementation with form validation
- The database migration looks appropriate for the feature

One minor suggestion for improvement: Consider adding a more descriptive comment in the frontend EditAgentForm component explaining the different states and behaviors (e.g., when it creates a new version vs. updates an existing one).

Please update the PR description with the required information, and this should be ready for review again.

File: autogpt_platform/frontend/src/app/(platform)/profile/(user)/dashboard/components/MainDashboardPage/useMainDashboardPage.ts [hunk 1]
```diff
@@ -59,6 +75,41 @@ export const useMainDashboardPage = () => {
...
+    setEditState({
+      isOpen: true,
+      submission,
+    });
+  };
+
+  const onEditSuccess = async (submission: StoreSubmission) => {
+    try {
+      if (!submission.store_listing_version_id) {
+        console.error("No store listing version ID found for submission");
```
Comment (0ubbe) [inline autogpt_platform/frontend/src/app/(platform)/profile/(user)/dashboard/components/MainDashboardPage/useMainDashboardPage.ts]: Let's send this error and the one below to Sentry instead:
```
import * as Sentry from "@sentry/nextjs"
// ...
Sentry.captureException(error) 
```

File: autogpt_platform/backend/backend/server/v2/store/routes.py [hunk 2]
```diff
@@ -564,6 +564,54 @@ async def create_submission(
...
+        submission_request (StoreSubmissionRequest): The updated submission details
+        user_id (str): ID of the authenticated user editing the listing
+
+    Returns:
+        StoreSubmission: The updated store submission
+
+    Raises:
+        HTTPException: If there is an error editing the submission
+    """
+    try:
```
Comment (majdyz) [inline autogpt_platform/backend/backend/server/v2/store/routes.py]: Don't you think we can simply remove the try catch here?

**Review by 0ubbe — DISMISSED** [review]
Front-end looks good 🏁 ✅ ; let's have a review/approval from a Backender too

**Review by majdyz — APPROVED** [review]
Feel free to retag once you've done with the conflict resolution

**Comment by AutoGPT-Agent** [comment]
Thanks for this comprehensive PR adding editing functionality for agent submissions! The implementation looks solid and follows good practices.

I like how you've structured the backend logic to handle different behaviors based on submission status (creating a new version for APPROVED submissions vs. updating existing versions for PENDING ones). The frontend components are also well organized.

The e2e tests you've added are thorough and cover the different scenarios properly.

Before this can be merged:

1. The PR has a merge conflict that needs to be resolved (indicated by the 'conflicts' label)

2. In the agent-dashboard.spec.ts file, there appear to be some duplicated test blocks and misplaced code sections that should be cleaned up

Otherwise, this PR is in good shape. The code changes match the intention stated in the PR title and description, and you've included appropriate tests to verify the functionality works as expected.

Once the conflicts are resolved, this should be ready for final review and merging.

**Review by majdyz — APPROVED** [review]
